app/src/main/java/org/mozilla/fenix/ext/BrowserIcons.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ext
+
+import android.view.View
+import android.widget.ImageView
+import androidx.annotation.MainThread
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.runBlocking
+import mozilla.components.browser.icons.BrowserIcons
+import mozilla.components.browser.icons.IconRequest
+import java.util.concurrent.CancellationException
+
+/**
+ * Loads an icon asynchronously using [BrowserIcons] and then displays it in the [ImageView].
+ */
+@MainThread
+fun BrowserIcons.loadInto(view: ImageView, request: IconRequest) {
+    val deferredIcon = loadIcon(request)
+    view.addOnAttachStateChangeListener(CancelOnDetach(deferredIcon))
+
+    deferredIcon.invokeOnCompletion {
+        try {
+            val icon = runBlocking { deferredIcon.await() }
+            view.setImageBitmap(icon.bitmap)
+        } catch (e: CancellationException) {
+            view.setImageDrawable(null)
+        }
+    }
+}
```

Comment:
I wonder if this will have issues with view recycling e.g. in RecyclerView?
* If a view is getting recycled (e.g. while scrolling) will we make sure only the last icon gets loaded? Afaik other libraries make use of a view `tag` to make sure the completed load is still something we want to display (or cancel a previous load)
* Should we remove the listener? Especially with view recycling wouldn't we pile up multiple listeners that never get cleared?